module/modmanager/manager.go

```diff
@@ -143,30 +145,84 @@ func (mgr *Manager) Handles() map[string]modlib.HandlerMap {
 	return res
 }
 
-// Add adds and starts a new resource module.
-func (mgr *Manager) Add(ctx context.Context, conf config.Module) error {
+func (mgr *Manager) lockModuleMutex(modName string) func() {
 	mgr.mu.Lock()
 	defer mgr.mu.Unlock()
-	return mgr.add(ctx, conf)
+
+	value, _ := mgr.perModMu.LoadOrStore(modName, &sync.Mutex{})
+	mu := value.(*sync.Mutex)
+	mu.Lock()
+	return mu.Unlock
 }
 
-func (mgr *Manager) add(ctx context.Context, conf config.Module) error {
+// Add adds and starts a new resource module.
+func (mgr *Manager) Add(ctx context.Context, confs ...config.Module) error {
 	if mgr.untrustedEnv {
 		return errModularResourcesDisabled
 	}
 
-	_, exists := mgr.modules[conf.Name]
-	if exists {
-		return nil
+	var (
+		wg      sync.WaitGroup
+		errs    = make([]error, len(confs))
+		newMods = make([]*module, len(confs))
+	)
+
+	func() {
```

Comment:
[q] Why put this in an anonymous function and run it immediately?

Reply:
my mistake, I used to have a lock here that was scoped to the function - I changed the lock scope but forgot to remove this function